libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/ITwilioClient.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Adapters.Twilio
+{
+    public interface ITwilioClient
+    {
+        void LogIn(string username, string password);
+
+        Task<string> GetResourceIdentifier(object messageOptions);
```

Comment:
Why are we using `object` here?

Reply:
My idea was to make the interface the most Twilio rest API agnostic possible. I didnt want the interface to depend on the current API (there would be almost no point on using an interface).

Now that I think about it, maybe we can remove the interface and use directly the `TwilioApi` class?  But I dont like the idea of the adapter depending on that class.

Reply:
Yeah, you're right. I think that the _not depending on Twilio_ is a good approach. Let's leave this class as you said, but keep in mind that we might be changing the _object_ type with another type (I don't know with which one, yet)

Reply:
Maybe with an adapter and an interface.
I would suggest using an interface instead of `CreateMessageOptions` but we cannot modify, so maybe wrap that class in a custom class and that one implement the interface. But seems kind of overkill.